src/node/node_info_network.h

```diff
@@ -11,14 +11,38 @@ namespace ccf
 {
   struct NodeInfoNetwork
   {
-    std::string rpchost;
-    std::string pubhost;
-    std::string nodehost;
-    std::string nodeport;
-    std::string rpcport;
-    std::string pubport;
+    struct NetAddress
+    {
+      std::string hostname;
+      std::string port;
+
+      bool operator==(const NetAddress& other) const
+      {
+        return hostname == other.hostname && port == other.port;
+      }
+    };
+
+    struct RpcAddresses
+    {
+      NetAddress rpc_address;
+      NetAddress public_rpc_address;
```

Comment:
I keep going back-and-forth on whether this should be here. We never use the public address of anything but the first interface, so maybe we just need 1 public address. Its up to the operator how the locally-bound interfaces are exposed on the public internet, and then they need to give the node a single name for this for redirection/node info purposes? But if we end up segregating specific frontends or endpoints on each interface, we want finer control of public interfaces ("if you're a user looking for node 0, the public address is FOO, if you're a member/looking for operator functionality, its BAR").

Reply:
Different interfaces may be routed differently, in particular if we ever get to multiple apps and more complex policy rules. I agree for now that it's largely just one interface that will have a public variant, but it seems reasonable to keep it specific just in case?